src/plugins/telemetry/server/plugin.ts

```diff
@@ -131,18 +131,16 @@ export class TelemetryPlugin implements Plugin<TelemetryPluginSetup, TelemetryPl
     };
   }
 
-  public async start(core: CoreStart, { telemetryCollectionManager }: TelemetryPluginsDepsStart) {
+  public start(core: CoreStart, { telemetryCollectionManager }: TelemetryPluginsDepsStart) {
     const { savedObjects, uiSettings, elasticsearch } = core;
     this.savedObjectsClient = savedObjects.createInternalRepository();
     const savedObjectsClient = new SavedObjectsClient(this.savedObjectsClient);
     this.uiSettingsClient = uiSettings.asScopedToClient(savedObjectsClient);
     this.elasticsearchClient = elasticsearch.client;
```

Comment:
1. I am slightly worried about any side affects that might result from not awaiting `handleOldSettings`. Telemetry might be enabled/disabled inside this function yet if we are not awaiting it other functions might not pick up the right value. What do you think?



2. Can we move all the above (`savedObjectsClient`  and `uiSettingsClient` from line 136) to the `handleOldSettings` function? Since it is an async function these sync methods will not block the main thread when called.

Reply:
Yeah! Makes sense! I think we had all the private clients because we used to instantiate the collectors in this plugin. But now they live in `kibana_usage_collection`, so it makes sense to only use them where appropriate.

I'll push an update in a sec!